### ISSUE TITLE
Animations should respect user preferences

### ISSUE THREAD

**Issue by LukasFritzeDev** [mittwald/flow#1991]
### Flow Version

current (Flow-Documentation)

### Bug Description

Even if prefers reduced motion is set in the OS or browser animations are added and played.

### Steps to Reproduce

1. Open Flow documentation, i.e. https://mittwald.github.io/flow/03-components/overlays/modal
2. Activate prefers reduced motion in OS settings or browser dev tools
3. Open a offcanvas modal

### Expected Behavior

> The prefers-reduced-motion CSS media feature is used to detect if a user has enabled a setting on their device to minimize the amount of non-essential motion. The setting is used to convey to the browser on the device that the user prefers an interface that removes, reduces, or replaces motion-based animations.

> Such animations can trigger discomfort for those with [vestibular motion disorders](https://www.a11yproject.com/posts/understanding-vestibular-disorders/). Animations such as scaling or panning large objects can be vestibular motion triggers.

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion

If the setting is set by the user, animations should be deactivated.

### Actual Behavior

Animations are always used. User preferences are not respected. 

### How much does this bug impact you?

🟡 Moderate – It's annoying but there's a workaround

### Relevant Logs or Error Output

```shell

```

### Browser Version

Safari 18.6

### Before Submitting

- [x] I have checked that this issue has not already been reported.
- [x] I have provided all necessary information to reproduce the issue.
